File: solution/1161. Maximum Level Sum of a Binary Tree/1161.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        int res = 1, mx = root->val, level = 1;
        queue<TreeNode*> q;
        q.push(root);
        while (!q.empty()) {
            int size = q.size();
            int sum = 0;
            while (size--) {
                auto node = q.front(); q.pop();
                sum += node->val;
                if (node->left) q.push(node->left);
                if (node->right) q.push(node->right);
            }
            if (sum > mx) {
                mx = sum;
                res = level;
            }
            level++;
        }
        return res;
    }
};

File: solution/1161. Maximum Level Sum of a Binary Tree/1161.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        res, mx, level = 1, root.val, 1
        q = collections.deque()
        q.append(root)
        while q:
            sum = 0
            for _ in range(len(q)):
                node = q.popleft()
                sum += node.val
                if (node.left): q.append(node.left)
                if (node.right): q.append(node.right)
            if sum > mx:
                mx, res = sum, level
            level += 1
        return res
